Tools/build/check_warnings.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+"""
+Parses compiler output with -fdiagnostics-format=json and checks that warnings exist
+only in files that are expected to have warnings.
+"""
+import argparse
+import json
+import re
+import sys
+from pathlib import Path
+
+def extract_warnings_from_compiler_output(compiler_output: str) -> list[dict]:
```

Comment:
This looks almost good to merge, just some style nits :)

Technically we're outside the stdlib so strictly speaking [PEP 8](https://peps.python.org/pep-0008/) doesn't apply, but let's follow it anyway. For example, please could you add extra newlines before these defs and wrap things to 79 chars?

The easiest way is to run Black (which we don't run on the stdlib but again we're outside the stdlib :)

```
black --line-length 79 Tools/build/check_warnings.py
```
Then some comments and long strings will need manual wrapping.

```suggestion

def extract_warnings_from_compiler_output(compiler_output: str) -> list[dict]:
```

Also, please don't force push in this repo, it makes it easier to review new commits, and we squash merge everything at the end anyway:

> In order to keep the commit history intact, please avoid squashing or amending history and then force-pushing to the PR. Reviewers often want to look at individual commits.

https://devguide.python.org/getting-started/pull-request-lifecycle/#quick-guide

Thanks!

Reply:
Updated to adhere to PEP8.

After a rebase there were several commits that came from branches from different forks. Not really sure how that happened but I had to revert to my last commit and force push to get rid of them.

Reply:
Thank you! I just pushed a commit to also wrap comments and strings to 79 chars as well.